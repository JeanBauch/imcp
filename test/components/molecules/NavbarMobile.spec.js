import { shallowMount, RouterLinkStub } from '@vue/test-utils';

import NabbarMobile from '~/components/molecules/NavbarMobile.vue';

describe('Components:molecules:NavbarMobile', () => {
	const factory = (values = {}) => {
		return shallowMount(NabbarMobile, {
			data () {
				return {
					...values,
				};
			},
			stubs: {
				RouterLink: RouterLinkStub,
			},
		});
	};
	const arrayToDataHeader = [
		{
			name: 'Início',
			link: 'index',
		},
		{
			name: 'Instituto',
			link: 'utilites',
		},
		{
			name: 'Escola de Música',
			link: 'utilites',
		},
		{
			name: 'Midia',
			link: 'utilites',
		},
		{
			name: 'Noticias',
			link: 'noticias',
		},
		{
			name: 'Agenda',
			link: 'utilites',
		},
		{
			name: 'Contato',
			link: 'contato',
		},
		{
			name: 'Transparência',
			link: 'utilites',
		},
	];
	describe('Snapshots', () => {
		it('should render component', async () => {
			const wrapper = factory();

			await wrapper.setData({ dataHeader: arrayToDataHeader });

			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
