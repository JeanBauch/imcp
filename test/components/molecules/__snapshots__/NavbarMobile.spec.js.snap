// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:molecules:NavbarMobile Snapshots should render component 1`] = `
"<nav id=\\"navList\\" class=\\"h-0 container mx-auto items-center justify-start px-6 sm:px-9 rounded-b-3xl transition-all ease-in-out duration-700 md:hidden\\">
  <ul class=\\"flex flex-col gap-2 border-t-2 border-gray border-opacity-20 w-full pt-4 invisible opacity-0\\">
    <li class=\\"text-white hover:font-semibold\\"><a>
        Início
      </a></li>
    <li class=\\"text-white hover:font-semibold\\"><a>
        Instituto
      </a></li>
    <li class=\\"text-white hover:font-semibold\\"><a>
        Escola de Música
      </a></li>
    <li class=\\"text-white hover:font-semibold\\"><a>
        Midia
      </a></li>
    <li class=\\"text-white hover:font-semibold\\"><a>
        Noticias
      </a></li>
    <li class=\\"text-white hover:font-semibold\\"><a>
        Agenda
      </a></li>
    <li class=\\"text-white hover:font-semibold\\"><a>
        Contato
      </a></li>
    <li class=\\"text-white hover:font-semibold\\"><a>
        Transparência
      </a></li>
  </ul>
</nav>"
`;
